gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiServiceImpl.java

```diff
@@ -681,6 +689,99 @@ public Page<GenericApiEntity> findAll(
             );
     }
 
+    @Override
+    public boolean isSynchronized(ExecutionContext executionContext, GenericApiEntity genericApiEntity) {
```

Comment:
Should we move this function elsewhere? I'm afraid that we will face the same issue we currently have with the "legacy" `ApiService` which is so fat.

What about moving it in the `SynchronizationService`? We could rename it as `isApiSynchronized`. Or we could create a new service `ApiSynchronizationService` 🤷

Reply:
Yes, I agree. 
SynchronizationService are also used for by PlanService. 

ApiSynchronizationService Sounds good to me. 
Or what do you think of `ApiDeploymentService` as we now want more expose a deploymentState ? (because be need to call ../deployments to change its state )

cc @phiz71

Reply:
Could the code be moved into the same service that handles deployments?
(ApiStateService)